Package.swift

```diff
@@ -144,6 +144,10 @@ let package = Package(
         ),
     ],
     targets: packageCollectionsSigningTargets + [
+        .target(
+            name: "CSQLite3",
+            exclude: ["CMakeLists.txt"]
+        ),
```

Comment:
we should probably also express the fact it is a system dependency now? @neonichu ?

Reply:
This actually is somewhat problematic.  There is a `TSCclibc` dependency that is still present that drags in `CSQLite3` from t-s-c which duplicates this module and thus causes a build issue.  I am not sure how to resolve this properly.  Would you have any suggestions?

Reply:
We should rename the module at a minimum, but I agree with Tom, we should be able to make this a system library target, right? That would also mean we don't need the dummy C file.

Reply:
I don't think that we can make it a system library target.  That would break the build on Windows IIRC.  However, in the case that we _can_ make this work where it can be a system library target *and* the user can simply provide the location where the headers and linker content is available, I think that would be great.  I would prefer that as well.

Reply:
in practice this _is_ a system dependency, as we are not vendoring it here. as such we should describe it in the manifest accordingly which will use the pkg-config code to Mae sure it is present and emit a warning if not

Reply:
to be clear, I would also be very happy _not_ to have this system dependency, although iirc LLBuild brings it in anyways

Reply:
Sure, I understand that, but I am concerned it may also mean that the ability to build SPM with SPM on Windows with a custom build of SQLite is lost.  My concern is purely that we do not lose that ability as the only supported option for SPM development on Windows is with SPM.  The CMake build is only supported for building the toolchain image.

Reply:
sure, that seems reasonable. iirc with pkg-config you can inject additional paths etc but I have no idea about windows.

Reply:
So this sounds to me as if system library targets aren't supported at all or don't work well on Windows? If that's the case, it sounds reasonable to me to *not* make this one right now, but we should probably separately discuss what we can do to make system libraries a viable choice on Windows.

Reply:
Okay, I have a reasonable enough workaround - namely we still have to wrap the header.  That allows for the header search path to be honoured.  Without that the modulemap will need to be injected into the installation which seems a bit too invasive.  We can just use the `systemLibrary` on all targets afterall.

Reply:
The macOS build is still an issue.  Any recommendations on how to handle that?  Once macOS is fixed, fixing the windows build should be easy (it should just be two additional parameters to the CMake invocation).

Reply:
yea how about we name it "SwiftPMSQLite3" or something along these lines so it will namespaced?